### ISSUE TITLE
Manually add missing icons  / art

### ISSUE THREAD

**Issue by dskw** [GDX-X/PFS-BatchKit-Manager#70]
Hi,

I've used the batchkit to successfully install my PS2 and PS1 games on the internal hdd of my PS2 fat.
Some of the titles do not show a proper icon though, like SLES_013.00.The Unholy War

I was wondering how I can specify an icon on my own, or where the tools even get the icons from.

**Comment by GDX-X** [comment]
Hi,

Glad it works!

are you talking about the icons for hdd-osd or the ART covers for OPL?

**Comment by dskw** [comment]
hdd-osd
So far I liked that more than using OPL

The mentioned title in a PS1 game btw, but I also have the issue with some EU PS2 games not showing a proper icon.
The games seem to work though, although I didn't have time to test them all yet

**Comment by dskw** [comment]
I'd like to create the missing icons using ticky/ps2iconsys toolkit, but while I see that artworks for OPL are available through the ART folder I do not see where the batch tools cache their icons.
Is it done somewhere in a temporary or user folder while the script executes and where does it grab the files from?
How would I best inject my own files into the procedure?

**Comment by GDX-X** [comment]
Yes, I also use Milkshape. I uploaded a template of the 3D icons from PS1 games—you might have found it, I think. 😅

It's not a legal issue; it's mostly because of the file size. I was worried that GitHub might reject it since it's quite large—400 MB.

Regarding PS2 games, I use the save game icons. I've made quite a lot of them, but some are still missing, mostly from Japanese games.

**Comment by dskw** [comment]
Yes, that is essentially what I was doing just now:
Going into each game, create a savegame, copy the files via wLaunchElf to USB and then use the icon in the repo.

Github allows a repo size of up to 1GB without any concerns. Even above is tolerated.
Once you reach 5GB you might receive a mail from them asking to take corrections.
https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#repository-size-limits

A lot of the games can probably share their icons across regions.
For example the LOTR games have 4-5 region versions each and I bet it would not hurt to make them all reference the same icon files.

I see multiple ways to achieve this:
1. Soft links between folders on the file system, but this could be come problematic with portability. Not sure if they can be relative in windows and if they still work after decompression
2. A big mapping file at the root of the repo to redirect to specific folders for the content of each gameid
3. A simple text file with a predefined name in a gameid folder that contains the name of the actual gameid it references.

All have pros and cons, but I think option 3. mit be the easiest:
Check if the file (example: _REDIRECT) exits and if it does, read its content.
The content could simply be another gameid like SLES_553.47
Then use the content of that other gameid instead.

Any other file in the folder next to _REDIRECT could be used on top, to override the files of the referenced gameid, which would allow for things like custom icon.sys while still referencing another gameid's icon.

This way the repo will probably be much smaller and well below the 1GB, even if more games are added over time :)
What do you think?
